dsfs/dataset.go

```diff
@@ -23,7 +23,6 @@ import (
 // LoadDataset reads a dataset from a cafs and dereferences structure, transform, and commitMsg if they exist,
 // returning a fully-hydrated dataset
 func LoadDataset(store cafs.Filestore, path string) (*dataset.Dataset, error) {
-	path = PackageFilepath(store, path, PackageFileDataset)
```

Comment:
I'm having trouble understanding how the functionality is changing here. Is the only difference that the `path` value is unaffected for when it gets passed to NewDatasetRef on line 68?

Part of the problem is that none of the tests exercise this. Would you be able to add a test that shows the bad functionality (path ending in "/dataset.json") when this function uses the old version, that then gets fixed by this change? Looks like this would require changing PackageFilepath to handle the "mem" prefix. While you're there, could you clarify the comment to better explain what it's doing, the current comment makes no sense to me at all.

Reply:
The only difference is that path is unaffected when we pass it down to LoadDatasetRefs, which means we have a clean `path` (without `/dataset.json`) for when we need to add the path into `ds`.

To you testing point, yes that is a good addition to this pr!